core/server/common/src/main/java/alluxio/Server.java

```diff
@@ -11,23 +11,47 @@
 
 package alluxio;
 
+import org.apache.thrift.TProcessor;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+
 /**
- * Interface representing an Alluxio Server.
+ * Interface representing an Alluxio server.
  */
 public interface Server {
 
   /**
+   * @return the server dependencies
+   */
+  Set<Class<?>> getDependencies();
+
+  /**
+   * @return the server's name
+   */
+  String getName();
+
+  /**
+   * @return a map from service names to {@link TProcessor}s that serve RPCs for this server
+   */
+  Map<String, TProcessor> getServices();
+
+  /**
    * Starts the Alluxio server.
    *
-   * @throws Exception if the server fails to start
+   * @param options the start options
+   * @throws IOException if the server fails to start
    */
-  void start() throws Exception;
+  void start(Object options) throws IOException;
```

Comment:
I would prefer to parameterize the start options type for type safety and self-documenting code. This class would become `Server<T>`